src/Test/TestJavier.java

```diff
@@ -0,0 +1,202 @@
+package Test;
+
+import java.util.ArrayList;
+
+import CommandingOfficers.Commander;
+import Engine.Army;
+import Engine.GameInstance;
+import Engine.GameScenario;
+import Engine.Combat.CombatContext;
+import Engine.Combat.StrikeParams.BattleParams;
+import Engine.UnitMods.UnitModifier;
+import Engine.Combat.CombatContext.CalcType;
+import Terrain.MapInfo;
+import Terrain.MapLibrary;
+import Terrain.MapMaster;
+import Terrain.TerrainType;
+import Terrain.Maps.MapReader;
+import Units.UnitContext;
+import Units.UnitModel;
+
+public class TestJavier extends TestCase
+{
+  private Commander javierAttack;
+  private Commander javierDefend;
+  private MapMaster map;
+  private GameInstance game;
+
+  private void setupTest(MapInfo mapInfo)
+  {
+    GameScenario scn = new GameScenario();
+    Army[] armies = new Army[mapInfo.getNumCos()];
+    for( int i = 0; i < armies.length; ++i )
+      armies[i] = new Army(scn, CommandingOfficers.AW3.GE.Javier.getInfo().create(scn.rules));
+
+    javierAttack = armies[0].cos[0];
+    javierDefend = armies[1].cos[0];
+
+    map = new MapMaster(armies, mapInfo);
+    game = new GameInstance(armies, map);
+
+    turn(game);
+  }
+
+  @Override
+  public boolean runTest()
+  {
+
+    boolean testPassed = true;
+
+    testPassed &= validate(testNoTowerStats() , "  Javier 0T test failed!");
+    testPassed &= validate(testDoRTowerStats(), "  Javier DoR tower test failed!");
+    testPassed &= validate(testDSTowerStats() , "  Javier DS tower test failed!");
```

Comment:
I'm not sure how you get to the point of writing these lines without questioning your life choices.

Reply:
Javier specifically is less effort to write a math test than to do a playtest.
...and yeah, I'd question my life choices if I actually respected the game design behind Javier, but I don't.